### PR TITLE
[ADD][11.0] stock_return_request: New module

### PR DESCRIPTION
**Pull request by chienandalu**
This module allows to return stock from a location to either suppliers,
customers or internal locations reverting the corresponding pickings.

cc @Tecnativa TT20569

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Please put Travis in green.

File: stock_return_request/models/stock_return_request.py [hunk 1]
```diff
@@ -0,0 +1,517 @@
...
+        copy=False,
+    )
+    to_refund = fields.Boolean(
+        string="To refund",
+        readonly=True,
+        states={'draft': [('readonly', False)]},
+    )
+    show_to_refund = fields.Boolean(
+        compute='_compute_show_to_refund',
+        default=lambda self: 'to_refund' in self.env['stock.move']._fields,
```
Comment (rousseldenis) [inline stock_return_request/models/stock_return_request.py:106]: Why this as there is a compute function ?

File: stock_return_request/models/stock_return_request.py [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+
+    def action_confirm(self):
+        """Wrapper for multi. Avoid recompute as it delays the pickings
+           creation"""
+        with self.env.norecompute():
+            for one in self:
+                one._action_confirm()
+        self.recompute()
+
+    def _action_confirm(self):
```
Comment (rousseldenis) [inline stock_return_request/models/stock_return_request.py:234]: Maybe split this into smaller functions.

**Review by rousseldenis — COMMENTED** [review]
Code review. Minor remarks. LGTM

File: stock_return_request/models/stock_return_request.py (hunk 1 above)
Comment (chienandalu) [reply stock_return_request/models/stock_return_request.py:106]: If I remember well, it was to assure we had it visible with a record in a new state (not yet recorded)

**Comment by chienandalu** [comment]
Travis is happy now

File: stock_return_request/models/stock_move.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+        readonly=True,
+        store=True,
+    )
+
+    @api.depends('state', 'returned_move_ids', 'quantity_done',
+                 'reserved_availability', 'returned_move_ids.qty_returnable')
+    def _compute_qty_returnable(self):
+        """Looks for chained returned moves to compute how much quantity
+           from the original can be returned """
+        for one in self.filtered(lambda x: x.state not in ['draft', 'cancel']):
```
Comment (carlosdauden) [inline stock_return_request/models/stock_move.py]: if one is move better it's named as move?

File: stock_return_request/models/stock_return_request.py [hunk 4]
```diff
@@ -0,0 +1,517 @@
...
+        ], limit=1)
+        return warehouse
+
+    def _compute_show_to_refund(self):
+        for one in self:
+            if 'to_refund' not in self.env['stock.move']._fields:
+                continue
+            one.show_to_refund = True
+
+    def _create_picking(self, picking, picking_moves):
```
Comment (carlosdauden) [inline stock_return_request/models/stock_return_request.py]: Can you do this to pass extra vals avoiding extra write calls:
```suggestion
    def _create_picking(self, picking, picking_moves, extra_vals):
```

**Review by carlosdauden — CHANGES_REQUESTED** [review]

File: stock_return_request/models/stock_return_request.py (hunk 4 above)
Comment (carlosdauden) [reply stock_return_request/models/stock_return_request.py]: Better option is create _prepare_picking and use copy_data function:
https://github.com/odoo/odoo/blob/dd620c788fe46131db161449f1f988083d700aac/odoo/models.py#L3848

Comment (chienandalu) [reply stock_return_request/models/stock_return_request.py]: Sure! I went for the second choice. Thanks!

**Review by rafaelbn — APPROVED** [review]
:+1:  thanks

**Comment by pedrobaeza** [comment]
@carlosdauden @rousseldenis can you update your review, please?

**Review by carlosdauden — APPROVED** [review]
@rousseldenis comments

File: stock_return_request/models/stock_return_request.py (hunk 2 above)
Comment (chienandalu) [reply stock_return_request/models/stock_return_request.py:234]: Some parts already are. I'll keep it in my for future revisions.

**Review by rousseldenis — APPROVED** [review]

**Pull request merged**